sematic/api/endpoints/request_parameters.py

```diff
@@ -31,17 +33,90 @@
 ]
 
 
+@dataclass
+class SearchRequestParameters:
+    limit: int
+    order: Callable[[Any], Any]
+    cursor: Optional[str]
+    group_by: Optional[sqlalchemy.Column]
+    filters: Optional[BooleanClauseList]
+    include_fields: Optional[List[str]]
+
+
+def get_garbage_filters(request_args: Dict[str, str], supported_filters: List[str]) -> Tuple[bool, List[str]]:
+    filters_json: str = request_args.get("filters", "{}")
+    try:
+        filters: Dict = json.loads(filters_json)
+    except Exception as e:
+        raise ValueError(f"Malformed filters: {filters_json}, error: {e}")
+    
+    operand = list(filters.keys())[0]
+    contained_extra_filters = False
+
+    if operand in {"AND", "OR"}:
+        filters = cast(BooleanPredicate, filters)
+        operand = cast(Literal["AND", "OR"], operand)
+        garbage_filters = []
+        for filter_ in filters[operand]:
+            filter_name = list(filter_.keys())[0]
+            if filter_name not in supported_filters:
+                contained_extra_filters = True
+                continue
+            if filter_[filter_name] != {"eq": True}:
+                raise ValueError(f"The filter '{filter_name}' must use the predicate {'eq: true'}")
```

Comment:
I suspect this is needed:
```suggestion
                raise ValueError(f"The filter '{filter_name}' must use the predicate \{'eq: true'\}")
```